homeassistant/components/insteon/utils.py

```diff
@@ -381,15 +384,33 @@ def print_aldb_to_log(aldb):
 
 @callback
 def async_add_insteon_entities(
-    hass, platform, entity_type, async_add_entities, discovery_info
-):
-    """Add Insteon devices to a platform."""
+    hass: HomeAssistant,
+    platform: Platform,
+    entity_type: type,
+    async_add_entities: Callable,
+    discovery_info: dict[str, Any],
+) -> None:
+    """Add an Insteon group to a platform."""
     new_entities = []
-    device_list = [discovery_info.get("address")] if discovery_info else devices
+    address = discovery_info["address"]
+    device = devices[address]
+    for group in discovery_info["groups"]:
+        new_entities.append(entity_type(device=device, group=group))
+    async_add_entities(new_entities)
```

Comment:
```suggestion
    device = devices[discovery_info["address"]]
    async_add_entities(
        [entity_type(device=device, group=group) for group in discovery_info["groups"]]
    )
```
you could rewrite this as a list comprehension if you think that is more readable.

Reply:
or probably even omit the brackets `[]` and use is as a generator expression 🤔
```suggestion
    device = devices[discovery_info["address"]]
    async_add_entities(
        entity_type(device=device, group=group) for group in discovery_info["groups"]
    )
```